File: app/src/main/java/andiag/coru/es/welegends/persistence/DBHelper.java
package andiag.coru.es.welegends.persistence;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

/**
 * Created by Canalejas on 06/06/2016.
 */
public class DBHelper extends SQLiteOpenHelper {

    private static final String TAG = "DBHelper";
    static final String DB_NAME = "andiag_welegends.sqlite";
    static final int DB_VERSION = 2;

    public static final String SUMMONER_TABLE_NAME = "Summoners";

    public static final String _ID = "_id";
    public static final String SUMMONER_RIOT_ID = "riotId";
    public static final String SUMMONER_NAME = "name";
    public static final String SUMMONER_REGION = "region";
    public static final String SUMMONER_ICON_ID = "iconId";
    public static final String SUMMONER_LEVEL = "summonerLevel";
    public static final String SUMMONER_LAST_UPDATE = "lastUpdate";

    private static DBHelper dbHelper = null;

    private DBHelper(Context context) {
        super(context, DB_NAME, null, DB_VERSION);
    }

    public static DBHelper getInstance(Context context) {
        if (dbHelper == null) {
            dbHelper = new DBHelper(context);
        }
        return dbHelper;
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL("CREATE TABLE " + SUMMONER_TABLE_NAME + " (" +
                _ID + " INTEGER PRIMARY KEY AUTOINCREMENT, " +
                SUMMONER_RIOT_ID + " INTEGER, " +
                SUMMONER_ICON_ID + " INTEGER, " +
                SUMMONER_NAME + " TEXT, " +
                SUMMONER_REGION + " TEXT, " +
                SUMMONER_LEVEL + " INTEGER, " +
                SUMMONER_LAST_UPDATE + " INTEGER)");
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        db.execSQL("DROP TABLE IF EXISTS " + SUMMONER_TABLE_NAME);
        onCreate(db);
    }
}

File: WeLegends/app/src/main/java/es/coru/andiag/welegends/database/DBHelper.kt
package es.coru.andiag.welegends.database

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

/**
 * Created by Canalejas on 08/12/2016.
 */

class DBHelper private constructor(context: Context) : SQLiteOpenHelper(context, DBHelper.DB_NAME, null, DBHelper.DB_VERSION) {

    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL("CREATE TABLE " + SUMMONER_TABLE_NAME + " (" +
                _ID + " INTEGER PRIMARY KEY AUTOINCREMENT, " +
                SUMMONER_RIOT_ID + " INTEGER, " +
                SUMMONER_ICON_ID + " INTEGER, " +
                SUMMONER_NAME + " TEXT, " +
                SUMMONER_REGION + " TEXT, " +
                SUMMONER_LEVEL + " INTEGER, " +
                SUMMONER_LAST_UPDATE + " INTEGER)")
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        db.execSQL("DROP TABLE IF EXISTS " + SUMMONER_TABLE_NAME)
        onCreate(db)
    }

    companion object {
        private val TAG = "DBHelper"
        const val DB_NAME = "andiag_welegends.sqlite"
        const val DB_VERSION = 2

        const val SUMMONER_TABLE_NAME = "Summoners"

        val _ID = "_id"
        val SUMMONER_RIOT_ID = "riotId"
        val SUMMONER_NAME = "name"
        val SUMMONER_REGION = "region"
        val SUMMONER_ICON_ID = "iconId"
        val SUMMONER_LEVEL = "summonerLevel"
        val SUMMONER_LAST_UPDATE = "lastUpdate"

        private var dbHelper: DBHelper? = null

        fun getInstance(context: Context): DBHelper {
            if (dbHelper == null) {
                dbHelper = DBHelper(context)
            }
            return dbHelper as DBHelper
        }
    }
}
